src/main/java/org/assertj/core/api/SoftAssertion.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.function.Consumer;
+
+import org.opentest4j.MultipleFailuresError;
+
+/**
+ * Parent interface for soft assertion implementations.
+ * 
+ * @author Fr Jeremy Krieg
+ * @see AbstractSoftAssertions
+ * @see SoftAssertions
+ */
+public interface SoftAssertion extends ThrowableCollector {
```

Comment:
my renaming suggestion to follow up on your comment:
* `SoftAssertionProvider`
* `SoftAssertionExtensionPoint`
* `SoftAssertionExtension`

Nothing stands out to me, I could live with `ISoftAssertion` although I'm not fond of that pattern naming it makes it clear this is an interface.

Reply:
I think we should avoid the `*Extension*` variants as they are easy to confuse with the JUnit Jupiter concept of an "Extension" (indeed, we've already got `org.assertj.core.api.junit.jupiter.SoftAssertionsExtension`).

I don't mind the `ISoftAssertion` pattern of naming, but I prefer consistency and we haven't used that pattern elsewhere in the code.

So I think by a process of elimination that leaves us with `SoftAssertionProvider`. I actually quite like this.

Reply:
let's go with SoftAssertionProvider then

Reply:
Done. Actually, I went with the plural `SoftAssertionsProvider` because it seemed to fit better (more than one soft assertion is typically provided).